openverse-api/catalog/api/serializers/audio_serializers.py

```diff
@@ -0,0 +1,174 @@
+from rest_framework import serializers
+
+from catalog.api.controllers.search_controller import get_sources
+from catalog.api.models import AudioReport
+from catalog.api.serializers.media_serializers import (
+    _validate_enum,
+    MediaSearchQueryStringSerializer,
+    MediaSearchResultsSerializer,
+    MediaSerializer,
+    AboutMediaSerializer,
+)
+
+
+class AudioSearchQueryStringSerializer(MediaSearchQueryStringSerializer):
+    """ Parse and validate search query string parameters. """
+
+    """
+    Keep the fields names in sync with the actual fields below as this list is
+    used to generate Swagger documentation.
+    """
+    fields_names = [
+        *MediaSearchQueryStringSerializer.fields_names,
+        'source',
+        'categories',
+        'duration',
+    ]
+
+    source = serializers.CharField(
+        label="provider",
+        help_text="A comma separated list of data sources to search. Valid "
+                  "inputs: "
+                  f"`{list(get_sources('audio').keys())}`",
+        required=False
+    )
+    categories = serializers.CharField(
+        label="categories",
+        help_text="A comma separated list of categories; available categories "
+                  "include `music`, `podcast`, `audiobook`, and `news`.",
+        required=False
+    )
+    duration = serializers.CharField(
+        label='duration',
+        help_text="A comma separated list of audio lengths; available lengths "
+                  "include `short`, and `long`.",
+        required=False
+    )
+
+    @staticmethod
+    def validate_source(input_sources):
+        allowed_sources = list(get_sources('audio').keys())
+        input_sources = input_sources.split(',')
+        input_sources = [x for x in input_sources if x in allowed_sources]
+        input_sources = ','.join(input_sources)
+        return input_sources.lower()
+
+    @staticmethod
+    def validate_categories(value):
+        valid_categories = {
+            'music',
+            'podcast',
+            'news',
+            'audiobook',
+        }
+        _validate_enum('category', valid_categories, value)
+        return value.lower()
+
+    @staticmethod
+    def validate_duration(value):
+        valid_ratios = {  # TODO: Finalise duration filters
+            'short',
+            'long'
+        }
+        _validate_enum('aspect ratio', valid_ratios, value)
+        return value.lower()
+
+
+class AudioSerializer(MediaSerializer):
+    """ A single audio file. Used in search results."""
+
+    """
+    Keep the fields names in sync with the actual fields below as this list is
+    used to generate Swagger documentation.
+    """
+    fields_names = [
+        *MediaSerializer.fields_names,
+        'set',
+        'genre',
+        'duration',
+        'bit_rate',
+        'sample_rate',
+        'alt_files',
+        'detail_url',
+        'related_url',
+    ]
+
+    set = serializers.PrimaryKeyRelatedField(
```

Comment:
audio_set? Nitpicking, but this **is** shadowing the builtin `set`.

Reply:
This will also need a change in the models, making a separate PR for that.

Reply:
#133 contains the changes for models.

Reply:
Resolved in 798495d.